hazelcast-client/src/main/java/com/hazelcast/client/config/YamlClientFailoverConfigBuilder.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+import com.hazelcast.config.AbstractYamlConfigBuilder;
+import com.hazelcast.config.ConfigLoader;
+import com.hazelcast.config.InvalidConfigurationException;
+import com.hazelcast.config.yaml.YamlDomChecker;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.internal.yaml.YamlLoader;
+import com.hazelcast.internal.yaml.YamlMapping;
+import com.hazelcast.internal.yaml.YamlNode;
+import com.hazelcast.nio.IOUtil;
+import com.hazelcast.util.ExceptionUtil;
+import org.w3c.dom.Node;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Properties;
+
+import static com.hazelcast.config.yaml.W3cDomUtil.asW3cNode;
+
+/**
+ * Loads the {@link com.hazelcast.client.config.ClientFailoverConfig} using YAML.
+ */
+public class YamlClientFailoverConfigBuilder extends AbstractYamlConfigBuilder {
+
+    private final InputStream in;
+
+    public YamlClientFailoverConfigBuilder(String resource) throws IOException {
+        URL url = ConfigLoader.locateConfig(resource);
+        if (url == null) {
```

Comment:
why not reuse `Preconditions`?

Reply:
Same as above.

Reply:
Even if decided to throw `IllegalArgumentException`, I would still prefer a utility method instead of repeating this if statement. I see it is repeated 9 times in this PR.

Reply:
Replaced with `Preconditions.checkTrue()` in all config classes and the failover builders.